P/pandoc_crossref/utils.jl

```diff
@@ -0,0 +1,27 @@
+# pandoc-crossref appears to be using the Haskell Package Versioning Policy. The forth
+# component of the version number is maintainer specified and for pandoc-crossref is always
+# a digit followed by an optional letter.
+#
+# https://pvp.haskell.org/
+# https://github.com/lierdakil/pandoc-crossref/releases
+function pandoc_crossref_jll_version(v::AbstractString)
+    m = match(r"^(?<major1>\d+)\.(?<major2>\d{1,2})\.(?<minor>\d+)\.(?<patch>\d)(?<build>[a-z])?$", v)
+    if !isnothing(m)
+        if !isnothing(m[:build])
+            build_letter = only(collect(m[:build]))
+            build_num = build_letter - 'a' + 1
+            1 <= build_num <= 9 || throw(ArgumentError("Found unexpected pandoc-crossref build value in: $v"))
+        else
+            build_num = 0
+        end
+
+        # Note: Version 0.3.16.0a occurs after 0.3.16.0
+        return VersionNumber(
+            parse(Int, m[:major1]) * 100 + parse(Int, m[:major2]),
+            parse(Int, m[:minor]),
+            parse(Int, m[:patch]) * 10 + build_num,
```

Comment:
Is 10 enough?  What if they got past `l`?

Reply:
Historically the highest they ever have got to is `f`. There is a check in place to ensure that the build number is always a single numeric digit. I was thinking if they do suddenly go past `i` we could update the convention to use `* 100` and the new version would be larger than any version we can have with the current function `9i` is the current max.

Reply:
I dug more into this and there is a possible collision:
```
julia> pandoc_crossref_jll_version("0.3.17.1")  # `9i` max
v"3.17.10"

julia> pandoc_crossref_jll_version2("0.3.17.0j")  # `9z` max
v"3.17.10"
```